initrd/bin/kexec-save-default

```diff
@@ -129,6 +129,41 @@ if [ ! -r $ENTRY_FILE -o ! -r $HASH_FILE ]; then
 	die "Failed to write default config"
 fi
 
+# logic to parse OS initrd to get extract crypttab, its filepaths and its OS defined options
+mkdir -p /tmp/initrd_extract
+cd /tmp/initrd_extract
+# User just selected a default boot option. Get initrd filename
+current_default_initrd=$(cat /boot/kexec_default_hashes.txt | grep initr | awk -F " " {'print $NF'} | sed 's/\.\//\/boot\//g')
+# Get crypttab files paths from initrd
+crypttab_files=$(zcat $current_default_initrd | cpio --list --quiet | grep crypttab) > /dev/null || crypttab_files=$(cpio --list --quiet < $current_default_initrd | grep crypttab) > /dev/null
+
+if [ ! -z "$crypttab_files" ]; then
+	echo "Extracting current selected default boot's $current_default_initrd for crypttab analysis..."
+	zcat $current_default_initrd | cpio -id --quiet "$crypttab_files" > /dev/null || cpio -id --quiet < $current_default_initrd "$crypttab_files" > /dev/null
+
+	crypttab_entries=$(cat "$crypttab_files"|uniq| sed 's/none/\/secret.key/g')
```

Comment:
The logic is incorrect here, since multiple crypttab files (paths) should be kept to have heads recreate them in propoer paths, not disturbing OS behavior. 

I think the problem here is `uniq|sed` (so uniq on filename, not on sed output) instead of `sed/uniq`. Will change for review. Otherwise I would have to install PureOS to test (currently testing on real hardware with Qubes (no crypptab in initrd) and debian-11-xfce on qemu.

----

I agree, OS maker should only have one crypptabl file in initrd, but this whole PR goal is to not care about OS choices, just respecting them and to replace `none` with `/secret.key` in OS's initrd expected path (overwriting them in memory) with OS's expected content (hoping OSes have propoer LUKS unlock logic, which they should if they support LUKS partitioning scheme at install).

Reply:
The problem is `cat "$crypttab_files"` - that tries to cat a single file whose name is the entire content of that variable, it's not word-split since it's quoted.

If you want to support it, do you want to loop here to apply the replacement to each file?  That would seem to make the fewest assumptions about equivalence, etc., but you'd have to store the new content for each location somewhere.

Maybe you could build the whole "crypttab override initrd" here rather than at boot time?  I.e. perform the replacement individually on each found file, put them all in a cpio archive, and save that single file rather than saving separate files listing the paths of the crypttabs and new contents.  Then boot would just cat the override initrd on, it wouldn't have to build it at all.